pkg/controllers/awsloadbalancercontroller/deployment.go

```diff
@@ -257,21 +267,23 @@ func (r *AWSLoadBalancerControllerReconciler) desiredDeployment(name, credential
 			})
 		}
 	}
-	return d
+	return d, nil
 }
 
-func desiredContainerArgs(controller *albo.AWSLoadBalancerController, clusterName, vpcID string) []string {
+func desiredContainerArgs(controller *albo.AWSLoadBalancerController, platformStatus *configv1.PlatformStatus, clusterName, vpcID string) ([]string, error) {
 	var args []string
 	args = append(args, fmt.Sprintf("--webhook-cert-dir=%s", webhookTLSDir))
 	args = append(args, fmt.Sprintf("--aws-vpc-id=%s", vpcID))
 	args = append(args, fmt.Sprintf("--cluster-name=%s", clusterName))
 
-	// if additional keys are present then sort them and append it to the arguments
-	if controller.Spec.AdditionalResourceTags != nil {
-		var tags []string
-		for _, t := range controller.Spec.AdditionalResourceTags {
-			tags = append(tags, fmt.Sprintf("%s=%s", t.Key, t.Value))
-		}
+	tags := mergeTags(controller, platformStatus)
+	// `--default-tags` arg allows a maximum of 24 user tags, but the combination of
+	// controller.Spec.AdditionalResourceTags and platformStatus.AWS.ResourceTags can result in more.
+	// Ensure a maximum of 24 merged tags are added.
+	if len(tags) > 24 {
```

Comment:
`24` is coming from validation added for the `AdditionalResourceTags` field. 


https://github.com/openshift/aws-load-balancer-operator/blob/59f061647f3d4869b0be0d111e784cf25dc9707f/api/v1/awsloadbalancercontroller_types.go#L61-L77

Since we are now merging these tags with the Infrastructure tags, isn't this number still valid?

Reply:
As per the discussion during the sync call, I've made the `MaxItems=11` for `AdditionalResourceTags` field to make room for platform tags. So the split of the total 50 tags will look like the following:
- ALBC : 3 tags
- PlatformStatus : 25 tags
- AdditionalResourceTags : 11 tags
- Ingress annotation: 11 tags

With this change, we don't need the above `if` block, which has been removed in the latest commit.